odriveS1/src/odriveS1/odrive.py

```diff
@@ -0,0 +1,36 @@
+from typing import ClassVar, Mapping, Any
+from typing_extensions import Self
+
+from viam.module.types import Reconfigurable
+from viam.proto.app.robot import ComponentConfig
+from viam.proto.common import ResourceName
+from viam.resource.base import ResourceBase
+from viam.resource.types import Model, ModelFamily
+
+from viam.components.motor import Motor
+from viam.logging import getLogger
+
+from odrive.enums import *
+from .odriveCAN import OdriveCAN
+from .odriveS1 import OdriveS1
+
+LOGGER = getLogger(__name__)
+MINUTE_TO_SECOND = 60
+
+class Odrive(Motor, Reconfigurable):
+    MODEL: ClassVar[Model] = Model(ModelFamily("viam-labs", "motor"), "odrive")
+    serial_number: str
+    max_rpm: float
+    odrive_config_file: str
+    odrv: Any
+
+    @classmethod
+    def new(cls, config: ComponentConfig, dependencies: Mapping[ResourceName, ResourceBase]) -> Self:
+        odrive = cls(config.name)
+        if config.attributes.fields["connection_type"].string_value == "canbus":
+            odrive.MODEL = ClassVar[Model] = Model(ModelFamily("viam-labs", "motor"), "odrive-can")
```

Comment:
the problem with this is that the user has to specify `odrive-can` as the model name in the config. the only reason this can't also just be `odrive` is because I'm adding both `OdriveCAN` and `OdriveS1` to the registry in the init file. If I can fix that problem then I will change this name back to just `odrive`

Reply:
I think it likely that we'll separate resource models as per OPTION 1 config in the scope doc. So this is actually good progress there.